### ISSUE TITLE
Change qdata:relevantDataSetFor to reference the Credential, Assessment, or Learning Opportunity

### ISSUE THREAD

**Issue by siuc-nate** [CredentialEngine/Schema-Development#755]
Per our 2-19-2020 meeting:

Since it doesn't make sense to make Holders Profile, Employment Outcome Profile, and/or Earnings Profile be top-level objects, we should enable qdata:DataSetProfiles to reference the relevant Credential, Assessment, and/or Learning Opportunity.

Proposal:
**Remove:**
> **Subject:** qdata:relevantDataSetFor
> **Predicate:** schema:rangeIncludes
> **Object:** ceterms:EarningsProfile, EmploymentOutcomeProfile

**Add:**
> **Subject:** qdata:relevantDataSetFor
> **Predicate:** schema:rangeIncludes
> **Object:** ceterms:Credential subclasses, ceterms:AssessmentProfile, ceterms:LearningOpportunityProfile

**Comment by stuartasutton** [comment]
So, may I assume from the "top-level" reference that this means that EarningsProfile and EmploymentOutcomeProfile are blank nodes? I question both why these are blank nodes or the conclusion that they have to reference the Credential subclasses, ceterms:AssessmentProfile, ceterms:LearningOpportunityProfile.

**Comment by siuc-nate** [comment]
> So, may I assume from the "top-level" reference that this means that EarningsProfile and EmploymentOutcomeProfile are blank nodes?

Currently, that is the case in the schema (though Mike had previously updated the API to require them to have CTIDs and enable them as top-level objects).  However, this creates a lot of extra top-level objects, CTIDs, URIs, etc., for publishers to manage (especially when they need to document all three kinds of aggregate data).  This is a significant and unnecessary hurdle given that we can rely on qdata:DataSetProfile to be a fully functional and fully-fledged top-level object capable of carrying aggregate data applicable to employment, earnings, and/or holders data.

> I question [...] why these are blank nodes

This eliminates the above problem while still letting them serve as "summary" objects for minimal information which are tightly-coupled to their host credential.

> I question [...] the conclusion that they have to reference the Credential subclasses, ceterms:AssessmentProfile, ceterms:LearningOpportunityProfile.

I think you may be thinking of the qdata:relevantDataSet property.  The issue/proposal here is to update the qdata:relevantDataSet**For** property to reference the Credential/Assessment/LearningOpportunity directly, rather than reference some Employment/Earnings/Holders profile that is a part of the data for that Credential/Assessment/LearningOpportunity.  

This proposal wouldn't change the Employment/Earnings/Holders profiles themselves, as qdata:relevantDataSetFor is a property of qdata:DataSetProfile.

In other words, instead of this:
![image](https://user-images.githubusercontent.com/21346519/108557098-a5254280-72bd-11eb-96a5-fd0ce622d5c2.png)

I am proposing this:
![image](https://user-images.githubusercontent.com/21346519/108557118-afdfd780-72bd-11eb-8bb6-a19c18eb4e82.png)

**Comment by stuartasutton** [comment]
Is there an assumption that a Credential can only have one EmploymenOutcomeProfile, HoldersProfile and Earnings profile?

**Comment by mparsons-ce** [comment]
@stuartasutton No, a credential can have multiple of each.

**Comment by mparsons-ce** [comment]
@siuc-nate re:
>Though Mike had previously updated the API to require them to have CTIDs and enable them as top-level objects

The latter was not an arbitrary decision on my part. It was based on discussions back in October.
https://docs.google.com/document/d/1bbO9tEcPcwKy8PMIBtb8_q_o9H6n0gZgOXw668brsfc/edit
We had a discussion with Stuart, regarding how the various objects should  published (see **Credential graph** tab)

https://app.diagrams.net/#G1afyg-UPz0uPytYn13AMPZsF5LCE_mu6C

re: proposal
The diagram could be interpreted as the DataSetProfile can be the relevantDataSetFor a credential without having any of the outcome profiles.

**Comment by philbarker** [comment]
I wasn't at the  2-19-2020 meeting so apologies if I'm dragging the conversation back.

I can see the pros and cons in Holders Profile, Employment Outcome Profile, and/or Earnings Profile being top-level objects. As Stuart mentioned in the CTDL meeting last week there are people who see only cons in blank nodes, and while that's a bit extreme, they can be a pain (the blank nodes, not the people).

But, I don't really see much rationale for linking a DataSetProfile directly to a Credential. I can see the obvious important reasons for wanting to get from the Holders Profile, Employment Outcome Profile, and/or Earnings Profile to the DataSetProfile, but is there a strong usecase for the opposite relationship that cannot be met with queries using relationships in the other direction? e.g. "show me all the Credentials that have a HoldersProfile with a relevantDataSet pointing at this one".

Also, the relationship between a DataSetProfile about a Credential (or more precisely about holders of a credential) seems different to the relationship between the summary outcome data and the dataset from which it is derived. I think this proposal changes the semantics of the property. Maybe consider `about` for this relationship, which is roughly what [DCAT](https://www.w3.org/TR/vocab-dcat-2/) does (I think), with its [theme](https://www.w3.org/TR/vocab-dcat-2/#Property:resource_theme) property that is [mapped](https://www.w3.org/TR/vocab-dcat-2/#dcat-sdo) sdo:about.

**Comment by siuc-nate** [comment]
> The latter was not an arbitrary decision on my part.

Sorry, I didn't mean to imply that it was. But thank you for the source documentation.

> The diagram could be interpreted as the DataSetProfile can be the relevantDataSetFor a credential without having any of the outcome profiles.

That was the intent. This is based on the ProPath case where all three of the CTDL aggregate data properties/profiles are just roundabout ways of getting to the same DataSetProfile, which holds information about earnings, outcomes, and employment.

> But, I don't really see much rationale for linking a DataSetProfile directly to a Credential.

The reasoning behind it that comes to my mind:
- The `qdata:relevantDataSetFor` property will need a range, and it doesn't make sense to let it point to things that don't have URIs
- Cases like the one referenced above where the DataSetProfile contains combined information about employment, earnings, and/or outcomes is something we've already seen in ProPath
- It enables, by default, third-party assertions along the lines of "this DataSetProfile is relevant data for this credential" which is another useful feature we will need eventually anyway.

Semantically speaking, wouldn't `schema:about` be intended to talk about the aggregate data itself (i.e., "This DataSetProfile is `about` this employment and earnings info")?

It would be really nice to consolidate the area of CTDL that talks about aggregate data into a single class and property, as it is very tedious to say/type "employment outcomes and earnings and holders" every time all three are in play, but that is maybe for a separate github issue.

**Comment by siuc-nate** [comment]
Note that [this issue](https://github.com/CredentialEngine/Schema-Development/issues/756) may influence our decisions here.

**Comment by philbarker** [comment]
> The reasoning behind it that comes to my mind:

>  - The qdata:relevantDataSetFor property will need a range, and it doesn't make sense to let it point to things that don't have URIs
> - Cases like the one referenced above where the DataSetProfile contains combined information about employment, earnings, and/or outcomes is something we've already seen in ProPath
>  - It enables, by default, third-party assertions along the lines of "this DataSetProfile is relevant data for this credential" which is another useful feature we will need eventually anyway.

The last one of those is more persuasive.

> Semantically speaking, wouldn't schema:about be intended to talk about the aggregate data itself (i.e., "This DataSetProfile is about this employment and earnings info")?

Yes. But something can be about more than one thing, and I assume it is not about all employment and earnings, it's about some credential's effect on employment and earnings. I wouldn't worry about saying census data for Kansas is data about Kansas, this seems analagous.

> It would be really nice to consolidate the area of CTDL that talks about aggregate data into a single class and property, as it is very tedious to say/type "employment outcomes and earnings and holders" every time all three are in play, but that is maybe for a separate github issue.

Yup.

**Comment by mparsons-ce** [comment]
I just noticed that I didn't post this last week.
Related.
If we will always have the relevantDataSet relationships, why would it be necessary to have relevantDataSetFor?

If the latter is a required relationship, then each time a new credential, lopp, or asmt wants to reference the datasetProfile, the datasetProfile would have to be republished. 
The DSP to credential connection can be derived via: DSP -> HP -> credential.

**However**, I don't see how multiple credentials can have the same dataSetProfile, especially if there are a lot of properties present in the DataProfile(s).
An argument may be where a credential is offered by the same university at five campuses, and each credential is published separately. I think the latter scenario is mostly discouraged. Ivy Tech and now Minnesota are examples where a single credential is published even if offered at multiple campuses.

**Comment by siuc-nate** [comment]
I believe relevantDataSetFor exists for RDF graph reasons, to be an inverse of relevantDataSet (hence why it currently points back to the Employment Outcome Profile, Holders Profile, and Earnings Profile).  But I wouldn't say that it is a required relationship at all.

You are probably right that DataSetProfiles will probably be related to one credential, but it's also reasonable that the DataSetProfile and the Credential may have different publishers (ie a state and a university, respectively).

**Comment by siuc-nate** [comment]
This issue will be handled in #756
